### PR TITLE
proxy access logs

### PR DESCRIPTION
**Pull request by benoitvidis**
prerequisite for https://github.com/kuzzleio/kuzzle-proxy/pull/44

Same implementation & changes as https://github.com/kuzzleio/kuzzle-plugin-websocket/pull/12

### CODE REVIEW

**Comment by benoitvidis** [comment]
Putting this PR in wip as I just realized the notifications could not be sent.

File: lib/index.js [hunk 1]
```diff
@@ -2,32 +2,24 @@
 
 module.exports = function SocketioProtocol () {
   this.config = {};
-  this.connectionPool = {};
+  this.sockets = {};
+  this.socketId2ConnectionId = {};
```
Comment (dbengsch) [inline lib/index.js]: nitpicking: (Personal feeling) I'm not really fond of digits in the middle of a var name

File: lib/index.js [hunk 2]
```diff
@@ -2,32 +2,24 @@
...
   this.protocol = 'socketio';
-  this.isDummy = false;
   this.io = null;
   this.context = null;
 
-  this.init = function (config, context, isDummy) {
+  this.init = function (config, context) {
     if (!config) {
```
Comment (sstafyniak) [inline lib/index.js]: Can we uniformize default configuration check like we've done in MQTT plugin ? 
https://github.com/kuzzleio/kuzzle-plugin-mqtt/blob/master/lib/index.js#L18

File: lib/index.js [hunk 3]
```diff
@@ -36,101 +28,97 @@ module.exports = function SocketioProtocol () {
...
+    let ips = [socket.handshake.address];
+    if (socket.handshake.headers['x-forwarded-for']) {
+      ips = socket.handshake.headers['x-forwarded-for']
+        .split(',')
+        .map(s => s.trim())
+        .concat(ips);
     }
 
-    this.connectionPool[socket.id] = socket;
+    try {
```
Comment (sstafyniak) [inline lib/index.js]: Huu, a `try` in another `try` ?

File: lib/index.js (hunk 2 above)
Comment (ballinette) [inline lib/index.js]: and also check the comment I made on this (while the PR has been merged meanwhile) : https://github.com/kuzzleio/kuzzle-plugin-mqtt/pull/8/files#r94320607

File: lib/index.js (hunk 3 above)
Comment (ballinette) [inline lib/index.js]: I don't see where you see an embed `try` ?

This one is at the `newConnection` method scope
The one below (line 86) is at the `socket.on(room)` event scope.

File: lib/index.js (hunk 2 above)
Comment (benoitvidis) [inline lib/index.js]: Sure, done.

File: lib/index.js (hunk 3 above)
Comment (stafyniaksacha) [inline lib/index.js]: ok, did not see that the second one is in `socket.on(this.config.room, data => {` method

But we register events listener inside try block ..

Comment (benoitvidis) [inline lib/index.js]: Why?

Anyway, I updated it.

**Comment by benoitvidis** [comment]
@sstafyniak can you review it please?
NB: This PR is quite urgent now as Kuzzle develop stack is broken till it is not merged.

**Pull request merged**